vendors/espressif/esp-idf/tools/cmake/idf_functions.cmake

```diff
@@ -113,6 +113,8 @@ function(idf_set_global_compile_options)
 
     list(APPEND compile_options "-Wall"
                                 "-Werror=all"
+                                "-Wno-error=char-subscripts"
```

Comment:
Can you please add this to relevant component? Modifying `vendors/espressif/esp-idf` is not good idea, it means divergence from our internal release, it will be difficult for us to keep track of such changes.

Reply:
okay, I'll see what I can do

Reply:
@mahavirj I have moved these to our CMakelist.txt and only added these compiler flags to the relevant files.